block/base/tx_priority.go

```diff
@@ -0,0 +1,168 @@
+package base
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type Coins = map[string]math.Int
+
+// DefaultTxPriority returns a default implementation of the TxPriority. It prioritizes
+// transactions by their fee.
+func DefaultTxPriority() TxPriority[string] {
+	return TxPriority[string]{
+		GetTxPriority: func(goCtx context.Context, tx sdk.Tx) string {
+			feeTx, ok := tx.(sdk.FeeTx)
+			if !ok {
+				return ""
+			}
+
+			return coinsToString(feeTx.GetFee())
+		},
+		Compare: func(a, b string) int {
+			aCoins, _ := coinsFromString(a)
+			bCoins, _ := coinsFromString(b)
+
+			switch {
+			case compareCoins(aCoins, bCoins):
+				return 1
+
+			case compareCoins(bCoins, aCoins):
+				return -1
+
+			default:
+				return 0
+			}
+		},
+		MinValue: "",
+	}
+}
+
+func coinsToString(coins sdk.Coins) string {
+	// sort the coins by denomination
+	coins.Sort()
+
+	// to avoid dealing with regex, etc. we use a , to separate denominations from amounts
+	// e.g. 10000,stake,10000,atom
+	coinString := ""
+	for i, coin := range coins {
+		coinString += coin.Amount.String() + "," + coin.Denom
+		if i != len(coins)-1 {
+			coinString += ","
+		}
+	}
+
+	return coinString
+}
+
+// coinsFromString converts a string of coins to a sdk.Coins object.
+func coinsFromString(coinsString string) (Coins, error) {
+	// split the string by commas
+	coinStrings := strings.Split(coinsString, ",")
+
+	// if the length is odd, then the given string is invalid
+	if len(coinStrings)%2 != 0 {
+		return nil, fmt.Errorf("invalid coins string: %s", coinsString)
+	}
+
+	coins := make(Coins)
```

Comment:
Can we preallocate capacity for this map since we know its length?
```suggestion
	coins := make(Coins, len(coinString))
```

Reply:
👍